Framework/CurveFitting/src/Algorithms/LeBailFit.cpp

```diff
@@ -471,31 +472,34 @@ void LeBailFit::execPatternCalculation() {
   // Calculate peak intensities and diffraction pattern
   vector<double> emptyvec;
   bool resultphysical = calculateDiffractionPattern(
-      m_dataWS->readX(m_wsIndex), m_dataWS->readY(m_wsIndex), true, true,
-      emptyvec, vecY, rfactor);
+      m_dataWS->x(m_wsIndex).rawData(), m_dataWS->y(m_wsIndex).rawData(), true,
+      true, emptyvec, vecY, rfactor);
+  m_outputWS->mutableY(CALDATAINDEX) = vecY;
 
   size_t numpts = vecY.size();
   // Calculate background
-  MantidVec &vec_bkgd = m_outputWS->dataY(INPUTBKGDINDEX);
+  std::vector<double> vec_bkgd(m_outputWS->y(INPUTBKGDINDEX).size(), 0);
   m_lebailFunction->function(vec_bkgd, vecX, false, true);
```

Comment:
Consider adapting the function signature of `LeBailFunction::function`. It does not seem to be used outside this algorithm so it would be a local change and would make your life easier here, I think.